catalog/catalog_products/views.py

```diff
@@ -0,0 +1,105 @@
+from typing import Dict
+from django.http.response import JsonResponse
+from rest_framework.generics import CreateAPIView
+from rest_framework.views import APIView
+from catalog_products.serializers import CategorySerializer, ProductSerializer
+from catalog_products.models import Category, Product
+from rest_framework import viewsets
+
+
+class CategoryAllViewSet(viewsets.ModelViewSet):
+    serializer_class = CategorySerializer
+    queryset = Category.objects.all()
+
+
+class ProductAllViewSet(viewsets.ModelViewSet):
+    serializer_class = ProductSerializer
+    queryset = Product.objects.all()
+
+
+class Task1ViewSet(APIView):
+    queryset = Category.objects.all()
+    serializer_class = CategorySerializer
+    def get(self, request):
+        category_all = Category.objects.all()
+        category_without_parent = []
+        category_with_parent = []
+        for category in category_all:
+            if category.parent != None:
+                category_with_parent.append(category)
+            else:
+                category_without_parent.append(category)
+        category_name_with_parent = list(map(lambda category: f'Subcategory: {category.name}, parent category: {category.parent.name}',category_with_parent))
+        category_name_without_parent = list(map(lambda category: f'Subcategory: {category.name} havent parent category!',category_without_parent))      
+        return JsonResponse(category_name_with_parent + category_name_without_parent, safe=False)
+
+
+class Task2ViewSet(APIView):
+    queryset = Category.objects.all()
+    serializer_class = CategorySerializer
+
+    def get(self, request):
+        def get_count_recoursively(category):
```

Comment:
I wonder how much requests does this function make?
Install django-debug-toolbar and see how much requests this function will make. 
I think that this approach will make a request per category with parent. So, f.e. if we had 100 categories with parent, we would end up with 100 same requests to the database.